src/auth/Header.tsx

```diff
@@ -1,16 +1,28 @@
-import { useNavigate } from 'react-router-dom';
+import { useState } from 'react';
+import { useLocation, useNavigate } from 'react-router-dom';
 import styled from 'styled-components';
+import { SidebarBackground } from '../layout';
+import Channels from '../layout/Channels';
+import Navbar from '../layout/Navbar';
 
 const Header = () => {
   const navigate = useNavigate();
+  const [menuOpen, setMenuOpen] = useState(false);
+  const pathname = useLocation().pathname.slice(1);
```

Comment:
```suggestion
  const {pathname} = useLocation();
```

이런 식으로 pathname 설정하고 pathname에 따라 channel이 달라지는 부분에서 주소를 `signup,`, `login` 이 아니라 `/signup`, `/login`으로 해주는 방법도 있을 것 같아요! 😄